lib/smack/SplitStructLoadStore.cpp

```diff
@@ -0,0 +1,137 @@
+#include "smack/SplitStructLoadStore.h"
+#include "llvm/IR/InstIterator.h"
+
+namespace smack {
+
+using namespace llvm;
+
+std::vector<Value*> getFirsts(std::vector<std::pair<Value*, unsigned>> lst) {
```

Comment:
Wait, shouldn't there be a more succinct way of doing this?
There must be a library function or something like that that helps you to extract first elements of this vector of pairs, no?
I googled it and this is one potentially nice hit: https://stackoverflow.com/questions/9094132/c-stdtransform-vector-of-pairs-first-to-new-vector